lib/adds.h

```diff
@@ -37,21 +46,21 @@ addsl2(long a, long b)
 
 
 inline long
-addsl3(long a, long b, long c)
+addslN(size_t n, long addend[n])
```

Comment:
Can we add a strict check that n >= 2? :)

I could see this ending up being called with n=1 and causing chaos.

Reply:
```c
if (n < 2) {
        errno = ENOTSUP; // or EDOM?
        return -1;
}
```

Something like this?

Reply:
For `n=1` we could still support it and return `addend[0]`.  And even returning `0` for `n=0` (nothing to add).  But that's probably a programmer bug, so let's errno out for now.

Reply:
Here's a proposal that accepts `n=1`, in case you prefer it.  I thought of sending that code before, since it removes the special case of n=2, but it adds one unnecessary qsort(3) call (with just two elements), which is why I didn't:

```c
diff --git a/lib/adds.h b/lib/adds.h
index c683d8c8..6544ce5f 100644
--- a/lib/adds.h
+++ b/lib/adds.h
@@ -50,8 +50,13 @@ addslN(size_t n, long addend[n])
 {
        int   e;
 
+       if (n == 0) {
+               errno = EDOM;
+               return 0;
+       }
+
        e = errno;
-       while (n > 2) {
+       while (n > 1) {
                qsort(addend, n, sizeof(addend[0]), cmpl);
 
                errno = 0;
@@ -60,7 +65,7 @@ addslN(size_t n, long addend[n])
                        return addend[0];
        }
        errno = e;
-       return addsl2(addend[0], addend[1]);
+       return addend[0];
 }
```

Reply:
Yeah that's good - simplicity wins out :)